aclk/agent_cloud_link.c

```diff
@@ -1280,24 +1280,31 @@ void *aclk_main(void *ptr)
     struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
     struct netdata_static_thread *query_thread;
 
-    if (!netdata_cloud_setting) {
-        info("Killing ACLK thread -> cloud functionality has been disabled");
-        static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
-        return NULL;
-    }
+#if defined( DISABLE_CLOUD ) || !defined( ENABLE_ACLK)
+    info("Killing ACLK thread -> cloud functionality has been disabled");
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
+    return NULL;
+#endif
 
     info("Waiting for netdata to be ready");
     while (!netdata_ready) {
         sleep_usec(USEC_PER_MS * 300);
     }
 
+    while (!netdata_cloud_setting) {
```

Comment:
So if cloud is disabled in the configuration, what's the plan to gracefully exit this loop? Do you cancel the thread and kill it?

Reply:
We leave it running, if an agent has the soft disable (`enabled = no` in the config) but the claimng script is run then it will update the config settings in memory and cause this loop to fall-through. Otherwise it will just sit there waiting forever.

Reply:
Is this a problem for graceful shutdown? Do we expect the thread to timeout during shutdown?